src/cli/export/export.go

```diff
@@ -37,3 +55,119 @@ func exportCmd() *cobra.Command {
 func handleExportCmd(cmd *cobra.Command, args []string) error {
 	return cmd.Help()
 }
+
+func runExport(
+	ctx context.Context,
+	cmd *cobra.Command,
+	args []string,
+	ueco utils.ExportCfgOpts,
+	sel selectors.Selector,
+	backupID, serviceName string,
+) error {
+	r, _, _, _, err := utils.GetAccountAndConnect(ctx, sel.PathService(), repo.S3Overrides(cmd))
+	if err != nil {
+		return Only(ctx, err)
+	}
+
+	defer utils.CloseRepo(ctx, r)
+
+	exportLocation := args[0]
+	if exportLocation == "" {
+		// This should not be possible, but adding it just in case.
+		exportLocation = control.DefaultRestoreLocation + dttm.FormatNow(dttm.HumanReadableDriveItem)
+	}
+
+	Infof(ctx, "Exporting to folder %s", exportLocation)
+
+	eo, err := r.NewExport(
+		ctx,
+		backupID,
+		sel,
+		utils.MakeExportConfig(ctx, ueco),
+	)
+	if err != nil {
+		return Only(ctx, clues.Wrap(err, "Failed to initialize "+serviceName+" export"))
+	}
+
+	expColl, err := eo.Run(ctx)
+	if err != nil {
+		if errors.Is(err, data.ErrNotFound) {
+			return Only(ctx, clues.New("Backup or backup details missing for id "+backupID))
+		}
+
+		return Only(ctx, clues.Wrap(err, "Failed to run "+serviceName+" export"))
+	}
+
+	// It would be better to give a progressbar than a spinner, but we
+	// have know way of knowing how many files are available as of now.
+	diskWriteComplete := observe.MessageWithCompletion(ctx, "Writing data to disk")
+	defer func() {
+		diskWriteComplete <- struct{}{}
+		close(diskWriteComplete)
+	}()
+
+	err = writeExportCollections(ctx, exportLocation, expColl)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func writeExportCollections(
+	ctx context.Context,
+	exportLocation string,
+	expColl []export.Collection,
+) error {
+	for _, col := range expColl {
+		folder := filepath.Join(exportLocation, col.BasePath())
```

Comment:
it's easy to confuse the `BasePath` here the fully prefixed `FullPath`.  You might want to add a comment that this only contains folder info.

Also, looking at https://github.com/alcionai/corso/blob/main/src/internal/m365/onedrive/export.go#L156, wouldn't the `basePath` contain the folder IDs, not their names?  Does this need to change to `LocationPath` instead of `FullPath().Folders`?

Reply:
You are right in that we need `LocationPath`, but that is not available for `RestoreCollections` which is what we have at this stage. That said, we get back names(at least of OneDrive and SharePoint) as the folder names in kopia are the names and not ids.